### PR TITLE
Added social shares

### PR DESCRIPTION
**Pull request by rpg600**

### CODE REVIEW

File: app/migrations/Version20170324172552.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Migrations;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
```
Comment (tgalopin) [inline app/migrations/Version20170324172552.php]: Can you remove the comments and `abortIf` like in others migrations?

File: src/AppBundle/Admin/SocialShareAdmin.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+class SocialShareAdmin extends AbstractAdmin
+{
+    protected function configureFormFields(FormMapper $formMapper)
+    {
+        $formMapper
+            ->add('name', null, [
+                'label' => 'Nom',
+            ])
+            ->add('type', 'choice', [
```
Comment (tgalopin) [inline src/AppBundle/Admin/SocialShareAdmin.php]: You should use FQCN for form types

File: src/AppBundle/Controller/SocialShareController.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    /**
+     * @Route("/jepartage/{slug}", name="app_social_share_show")
+     * @Method("GET")
+     */
+    public function showAction(SocialShareCategory $socialShareCategory): Response
+    {
+        return $this->render('social/show.html.twig', [
+            'socialShareCategory' => $socialShareCategory,
+            'socialShares' => $this->getDoctrine()->getRepository(SocialShare::class)->findBy(['socialShareCategory' => $socialShareCategory, 'published' => true], ['position' => 'ASC', 'createdAt' => 'DESC']),
```
Comment (tgalopin) [inline src/AppBundle/Controller/SocialShareController.php]: This line seems too long.

File: src/AppBundle/Entity/SocialShare.php [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+     * @ORM\Column(type="bigint")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(length=100)
+     *
+     * @Assert\NotBlank
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: Can you add MaxLength 100 ?

File: src/AppBundle/Entity/SocialShare.php [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    {
+        $this->defaultUrl = $defaultUrl;
+    }
+
+    public function getFacebookUrl(): ?string
+    {
+        return $this->facebookUrl;
+    }
+
+    public function setFacebookUrl(string $facebookUrl)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: `?string`

File: src/AppBundle/Entity/SocialShare.php [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+    {
+        $this->facebookUrl = $facebookUrl;
+    }
+
+    public function getTwitterUrl(): ?string
+    {
+        return $this->twitterUrl;
+    }
+
+    public function setTwitterUrl(string $twitterUrl)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: `?string`

File: src/AppBundle/Entity/SocialShare.php [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+    {
+        $this->description = $description;
+    }
+
+    public function getDefaultUrl(): ?string
+    {
+        return $this->defaultUrl;
+    }
+
+    public function setDefaultUrl(string $defaultUrl)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: `?string`

File: src/AppBundle/Entity/SocialShare.php [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+    {
+        $this->position = $position;
+    }
+
+    public function getDescription(): ?string
+    {
+        return $this->description;
+    }
+
+    public function setDescription(string $description)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: `?string`

File: src/AppBundle/Entity/SocialShare.php [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+    {
+        return $this->slug;
+    }
+
+    public function getPosition(): int
+    {
+        return $this->position;
+    }
+
+    public function setPosition(int $position)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: ?int

File: src/AppBundle/Entity/SocialShare.php [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+    {
+        return $this->id;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: ?string

File: src/AppBundle/Entity/SocialShare.php [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+    {
+        $this->twitterUrl = $twitterUrl;
+    }
+
+    public function getType(): ?string
+    {
+        return $this->type;
+    }
+
+    public function setType(string $type)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: ?string

File: src/AppBundle/Entity/SocialShare.php [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+    {
+        $this->type = $type;
+    }
+
+    public function getSocialShareCategory(): ?SocialShareCategory
+    {
+        return $this->socialShareCategory;
+    }
+
+    public function setSocialShareCategory(SocialShareCategory $socialShareCategory)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: ?SocialShareCategory

File: src/AppBundle/Entity/SocialShare.php [hunk 13]
```diff
@@ -0,0 +1,227 @@
...
+    {
+        $this->socialShareCategory = $socialShareCategory;
+    }
+
+    public function getMedia(): ?Media
+    {
+        return $this->media;
+    }
+
+    public function setMedia(Media $media)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: ?Media

File: src/AppBundle/Entity/SocialShare.php [hunk 14]
```diff
@@ -0,0 +1,227 @@
...
+    {
+        $this->media = $media;
+    }
+
+    public function isPublished(): bool
+    {
+        return $this->published;
+    }
+
+    public function setPublished(bool $published)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: ?bool

File: src/AppBundle/Entity/SocialShareCategory.php [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+     * @ORM\Column(type="bigint")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(length=100)
+     *
+     * @Assert\NotBlank
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShareCategory.php]: Can you add MaxLength 100 ?

File: src/AppBundle/Entity/SocialShareCategory.php [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        return $this->id;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShareCategory.php]: ?string

File: src/AppBundle/Entity/SocialShareCategory.php [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        return $this->slug;
+    }
+
+    public function getPosition(): int
+    {
+        return $this->position;
+    }
+
+    public function setPosition(int $position)
```
Comment (tgalopin) [inline src/AppBundle/Entity/SocialShareCategory.php]: ?int

File: app/Resources/views/social/list.html.twig [hunk 18]
```diff
@@ -0,0 +1,2 @@
+socialShareCategories: {{ dump(socialShareCategories) }}
```
Comment (mykiwi) [inline app/Resources/views/social/list.html.twig]: dump ?

File: src/AppBundle/Entity/SocialShare.php (hunk 10 above)
Comment (rpg600) [inline src/AppBundle/Entity/SocialShare.php]: Why ? this field is mandatory.

Comment (tgalopin) [inline src/AppBundle/Entity/SocialShare.php]: Yes but the form should be able to put null into it and handle the problem with the validator.

**Comment by rpg600** [comment]
@tgalopin changes done.

**Pull request merged**